src/coreComponents/linearAlgebra/interfaces/hypre/HypreUtils.cpp

```diff
@@ -46,6 +48,68 @@ HYPRE_Vector parVectorToVectorAll( HYPRE_ParVector const vec )
   }
 }
 
+HYPRE_Vector parVectorToVector( HYPRE_ParVector const vec, int const targetRank )
```

Comment:
Maybe we could use https://github.com/hypre-space/hypre/blob/907a2d07b64fe47bdde4540c54665c83ced83a2c/src/parcsr_mv/par_vector.c#L985 instead

Do we need the situation when `targetRank != 0`?

Reply:
It's needed to support serial direct solver (`umfpack`) in cases when a target physics may not be present on all MPI ranks (like poromechanics with over/underburden). Because we use `hypre_ParCSRMatrixToCSRMatrixAll` to collect the matrix to a single working rank, and it cannot collect it to a rank where it has 0 rows, we may have to select a working rank other than 0. Now, `umfpack` is only useful for small-scale debugging runs, but I think I also tried it as a coarse-level solver in my multiscale implementation. So ideally I would like not to drop support for `targetRank != 0`.

Reply:
That makes sense! I'd suggest here again to move this functionality to hypre, i.e., implement what is missing in hypre. But, I'll let you guys decide @castelletto1

Reply:
Having linear algebra functionality in hypre makes sense to me.